pyobvector/client/ob_vec_json_table_client.py

```diff
@@ -0,0 +1,776 @@
+            col_type_params = col_def.kind.expressions
+            col_type_params_list = []
+            col_nullable = True
+            col_has_default = False
+            col_default_val = None
+            for param in col_type_params:
+                if param.is_string:
+                    col_type_params_list.append(f"'{param.this}'")
+                else:
+                    col_type_params_list.append(f"{param.this}")
+            if len(col_type_params_list) > 0:
+                col_type_str += '(' + ','.join(col_type_params_list) + ')'
+            col_type_model = ObVecJsonTableClient.JsonTableMetadata._parse_col_type(col_type_str)
+            
+            for cons in col_def.constraints:
+                if isinstance(cons.kind, exp.DefaultColumnConstraint):
+                    col_has_default = True
+                    logger.info(f"############ create jtable ########### {str(cons.kind.this)}")
+                    col_default_val = str(cons.kind.this)
+                    if col_default_val.upper() == "NULL":
+                        col_default_val = None
+                elif isinstance(cons.kind, exp.NotNullColumnConstraint):
+                    col_nullable = False
+                else:
+                    raise ValueError(f"{cons.kind} constriaint is not supported.")
+            
+            if col_has_default and (col_default_val is not None):
+                # check default value is valid
+                col_type_model(val=self._calc_default_value(col_default_val))
+
+            if (not col_nullable) and col_has_default and (col_default_val is None):
+                raise ValueError(f"Invalid default value for '{col_name}'")
+
+            logger.info(
+                f"col_name={col_name}, col_id={col_id}, "
+                f"col_type_str={col_type_str}, col_nullable={col_nullable}, "
+                f"col_has_default={col_has_default}, col_default_val={col_default_val}"
+            )
+            new_meta_cache_items.append({
+                'jcol_id': col_id,
+                'jcol_name': col_name,
+                'jcol_type': col_type_str,
+                'jcol_nullable': col_nullable,
+                'jcol_has_default': col_has_default,
+                'jcol_default': col_default_val,
+                'jcol_model': col_type_model,
+            })
+            session.add(ObVecJsonTableClient.JsonTableMetaTBL(
+                user_id = self.user_id,
+                jtable_name = jtable_name,
+                jcol_id = col_id,
+                jcol_name = col_name,
+                jcol_type = col_type_str,
+                jcol_nullable = col_nullable,
+                jcol_has_default = col_has_default,
+                jcol_default = {
+                    'default': col_default_val,
+                }
+            ))
+            
+            col_id += 1
+        
+        try:
+            session.commit()
+            self.jmetadata.meta_cache[jtable_name] = new_meta_cache_items
```

Comment:
What if there are more than one threads accessing `jmetadata.meta_cache`?

Reply:
Concurrency safety is not necessary in this PR